### PR TITLE
Fix updating no of post after deleting a post and load-more button

### PR DESCRIPTION
**Pull request by individual-it**
This pull request makes the following changes:
- ~~fixes updating the post counter when deleting a post~~
- loading the correct amount of posts when a post was deleted and then more posts are loaded
- ~~UI end-to-end test infrastructure with tests for the fixes in this PR~~

Testing checklist:
- [x] 22 Scenarios about deleting posts and showing more posts, see .feature files in this PR. All tests are implemented in CI and are passing, see #1473

- [x] I certify that I ran my checklist

Fixes ushahidi/platform#3861 &  ~~ushahidi/platform#3326~~

Ping @ushahidi/platform

### CODE REVIEW

**Comment by individual-it** [comment]
I'm leaving the commit history for now, just for reference. I can squash them when the responsible people are happy with the changes

**Comment by individual-it** [comment]
fixed conflicts that appeared because of ushahidi/platform-client#1443.

File: test/ui/stepDefinitions/login.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const {Given} = require('cucumber');
+const {client} = require('nightwatch-api');
+
+Given('the administrator has logged in into the webUI', async function () {
```
Comment (phil-davis) [inline test/ui/stepDefinitions/login.js]:
```suggestion
Given('the administrator has logged in to the webUI', async function () {
```

File: test/ui/pageObjects/ushahidiPage.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Page Objects that contains very general elements, that appear everywhere
```
Comment (phil-davis) [inline test/ui/pageObjects/ushahidiPage.js]:
```suggestion
 * Page Object that contains very general elements, that appear everywhere
```

**Comment by individual-it** [comment]
The issue is, that the order of posts is random

**Comment by individual-it** [comment]
@phil-davis I've addressed your requests

File: test/ui/features/paginationOfPosts.feature [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        And the administrator loads more posts using the webUI
+        Then there should be <expected-listed-posts> posts listed on the webUI
+        And the post-count header on the webUI should show "<expected-count-header>"
+        And there should be no load-more button
+        Examples:
+            | amount | expected-listed-posts | expected-count-header |
+            | 60     | 60                    | 60 of 60 posts        |
+            | 52     | 52                    | 52 of 52 posts        |
+
+    Scenario Outline: show-more-posts-button should disappear when all posts are loaded (loading more posts twice)
```
Comment (phil-davis) [inline test/ui/features/paginationOfPosts.feature]:
```suggestion
    Scenario Outline: show-more-posts-button should still be visible when not all posts are loaded (loading more posts twice)
```

File: test/ui/features/paginationOfPosts.feature [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        Given <amount> posts have been created by the public with the title "civil unrest", and the content "description"
+        And the administrator has logged in to the webUI
+        And the administrator has opened the data-view on the webUI
+        When the administrator loads more posts using the webUI
+        And the administrator loads more posts using the webUI
+        Then there should be <expected-listed-posts> posts listed on the webUI
+        And the post-count header on the webUI should show "<expected-count-header>"
+        And there should be no load-more button
```
Comment (phil-davis) [inline test/ui/features/paginationOfPosts.feature]: `load-more button` and `show-more-posts-button` (in scenario outline description) seem to be the same button.
Use the same words for the button.
(search-replace through whole scenario...)

File: test/ui/features/paginationOfPosts.feature [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            | 52     | 52                    | 52 of 52 posts        |
+
+    Scenario Outline: show-more-posts-button should disappear when all posts are loaded (loading more posts twice)
+        Given <amount> posts have been created by the public with the title "civil unrest", and the content "description"
+        And the administrator has logged in to the webUI
+        And the administrator has opened the data-view on the webUI
+        When the administrator loads more posts using the webUI
+        And the administrator loads more posts using the webUI
+        Then there should be <expected-listed-posts> posts listed on the webUI
+        And the post-count header on the webUI should show "<expected-count-header>"
```
Comment (phil-davis) [inline test/ui/features/paginationOfPosts.feature]: Can we have another step like:
```
And there should be a load-more button visible
```

Comment (individual-it) [reply test/ui/features/paginationOfPosts.feature]: addressed in 6f489ca2

File: test/ui/features/paginationOfPosts.feature (hunk 3 above)
Comment (individual-it) [reply test/ui/features/paginationOfPosts.feature]: addressed in 6f489ca2

File: test/ui/features/paginationOfPosts.feature (hunk 4 above)
Comment (individual-it) [reply test/ui/features/paginationOfPosts.feature]: addressed in 4c5785d1

**Comment by rowasc** [comment]
hey @individual-it. Thanks for your contribution. As I see this, the change should not include a completely new testing framework to be included. I encourage you to use the testing setup that is already here. 
I would like to discuss the possibility of adding nightwatch since it's a great tool, but that should be done in a separate PR to ensure we handle it as a separate decision.

Thanks!
Romina

**Comment by individual-it** [comment]
@rowasc I've added the tests to make sure the fix does fix all cases for the user, specially because at the beginning there have been kind of two separate issues.
But you are right the whole test-setup can go into a separate PR, I will pull it our

**Comment by individual-it** [comment]
This PR now only contains the fix, the UI tests are moved to  #1473

**Pull request merged**